src/library/messages.cpp

```diff
@@ -34,7 +34,7 @@ std::ostream & operator<<(std::ostream & out, message const & msg) {
 void report_message(message const & msg0) {
     if (!get_global_ios().get_options().get_bool(name {"trace", "as_messages"}, false)) {
         // Print immediately. Still add as a message so that we get the error code correct.
-        get_global_ios().get_regular_stream() << msg0;
+        get_global_ios().get_diagnostic_stream() << msg0;
```

Comment:
Not sure if this is an uncontroversial change. It means that `lean Foo.lean | less` won't work anymore. But that is how every other compiler works, and it is essential to make sure that in code like `$(LEAN) --deps $< | python relative.py`, the actual output flows into the pipe, while error messages are presented to the user.